protocol/implementations/js/src/protocolDefinitions.ts

```diff
@@ -0,0 +1,121 @@
+        // alice sends RFQs, not receives them
+        'RFQ': {
+            '$actions': [],
+            // whoever received the RFQ that Alice sent, can write back a Quote to Alice
+            'Quote': {
+                '$actions': [
+                    {
+                        'who': 'recipient',
+                        'of': 'RFQ',
+                        'can': 'write'
+                    }
+                ],
+                // alice sends Orders, not receives them
+                'Order': {
+                    '$actions': [],
+                    // whoever received the order that Alice sent in response to a Quote in response to an RFQ, can write back an OrderStatus to Alice.
+                    'OrderStatus': {
+                        '$actions': [
+                            {
+                                'who': 'recipient',
+                                'of': 'RFQ/Quote/Order',
+                                'can': 'write'
+                            }
+                        ]
+                    }
+                }
+            }
+        }
+    }
+};
+
+
+export const pfiProtocolDefinition = {
+    'protocol': 'https://tbd.website/protocols/tbdex',
+    'types': {
+        'RFQ': {
+            'schema': 'https://tbd.website/protocols/tbdex/RequestForQuote',
+            'dataFormats': [
+                'application/json'
+            ]
+        },
+        'Quote': {
+            'schema': 'https://tbd.website/protocols/tbdex/Quote',
+            'dataFormats': [
+                'application/json'
+            ]
+        },
+        'Order': {
+            'schema': 'https://tbd.website/protocols/tbdex/Order',
+            'dataFormats': [
+                'application/json'
+            ]
+        },
+        'OrderStatus': {
+            'schema': 'https://tbd.website/protocols/tbdex/OrderStatus',
+            'dataFormats': [
+                'application/json'
+            ]
+        }
+    },
+    'structure': {
+        // anyone can write RFQ to a PFI's DWN
+        // no one can read RFQs from PFI's DWN (except the PFI itself)
+        'RFQ': {
+            '$actions': [
+                {
+                    'who': 'anyone',
+                    'can': 'write'
+                }
+            ],
+            // PFI is sending OUT quotes. no one should be writing Quotes to PFIs.
+            'Quote': {
+                '$actions': [],
+                // PFI who received an RFQ/Quote (i.e. Alice) can write Order to PFI's DWN
+                // no one can read Order from PFI's DWN (except the PFI itself)
+                'Order': {
+                    '$actions': [
+                        {
+                            'who': 'recipient',
+                            'of': 'RFQ/Quote',
+                            'can': 'write'
+                        }
+                    ],
+                    // PFI is sending OUT OrderStatus. no one should be writing OrderStatus to PFIs.
```

Comment:
> PFI is sending OUT OrderStatus.

How does Alice request an OrderStatus? or is it "push only" to alice?

Reply:
this is a great question! i would think that it's push only from PFI to Alice's DWN, and then Alice can read the orderStatus pushed to her DWN (like her "inbox"). right now i think of DWNs mailboxes, or maybe even messaging queues. i'd like @mistermoe @andresuribe87 to correct me if i'm wrong though.

Reply:
Yes, that's my same understanding!

One thing that's not currently supported is a way to fetch the `OrderStatus` on demand. Imagine that Alice has sent an `Order` to the PFI. After 2 minutes, she doesn't see any `OrderStatus` in her DWN. Is there a mechanism that she can use to "force" it?

Reply:
One option is to mark Alice as the `recipient` of an `OrderStatus`, and add an `$action` rule allowing recipients to read. If Alice does not have a publicly addressable DWN, she can read the `OrderStatus` from PFI's DWN.

```
'OrderStatus': {
  '$actions': [{
    who: 'recipient',
    of: 'RFQ/Quote/Order/OrderStatus',
    can: 'read'
  }]
}
```

Reply:
Visual representation... yeah def **push** the `OrderStatus` IMO

<img width="801" alt="image" src="https://github.com/TBD54566975/tbdex-protocol/assets/3602993/147a4fe9-8c43-47fc-adb3-8677f7fc71da">

Reply:
@KendallWeihe Yes a PFI can write multiple `OrderStatus`s with the definition above.

Reply:
@KendallWeihe how would a pull work? there would have the be a request for status message? 

Diagram is nice too - if it is helpful consider adding it to the readme of the repo perhaps?

Reply:
added diagram to readme, with the edit of OrderStatus being a push from PFI to Alice :)